relay-server/src/services/processor/replay.rs

```diff
@@ -262,13 +264,42 @@ fn handle_replay_recording_item(
     })
     .map(Into::into)
     .map_err(|error| {
+        match &error {
+            relay_replays::recording::ParseRecordingError::Compression(e) => {
+                // 20k errors per day at 0.1% sample rate == 20 logs per day
+                if rand::thread_rng().gen_range(1..=1000) == 1 {
```

Comment:
For this we have a sample function https://github.com/getsentry/relay/blob/4c0638ab8a20f22bd9e146ba49bdf81395f9d869/relay-server/src/utils/pick.rs#L9-L14